Week3/homework/step2-7.js

```diff
@@ -20,4 +20,28 @@ f2(y);
 
 console.log(y);
 
-// Add your explanation as a comment here
+/**
+ * Inside f1 function a new variable (val) is being created instead of x.
+ * We use x as an argument but we don't manipulate it.
+ * If we assign f1 function to a new variable, then we can get the expected value.
+ * /////////////////////////////////
+ * // const altX = f1(x)        ///
+ * // console.log(altX) --> 10 ///
+ * //////////////////////////////
+ */
+
+/**
+ * Reference types (object/array) are different from primitive types.
+ * Each object/array has a reference address in disk. If we reassign an object, its address will be changed.
+ * Then it will never be reassigned and its address will never be changed.
+ * But if we change a property of an object, then the object will be manipulated. But the address still will be the same.
+ * No matter in which scope we change the property, we get the manipulated object.
+ * Even if we assign our object to a new object, changing the property of the new object will effect the original object.
```

Comment:
Overall, I think you have a good understanding of the concepts of call by value and call by reference, but I'm not sure if I understand this line: 'Even if we...'

Reply:
> but I'm not sure if I understand this line: 'Even if we...'

I tried to explain this:

```javascript
  const objX = { name: "Frank" }
  cons objY = objX;

   objY.age = 20;
  console.log(objX); // { name: "Frank", age: 20 }
```